shell/common/rasterizer.cc

```diff
@@ -670,6 +678,27 @@ std::optional<size_t> Rasterizer::GetResourceCacheMaxBytes() const {
   return std::nullopt;
 }
 
+bool Rasterizer::EnsureThreadsAreMerged() {
+  // If threads are merged statically, always return true.
+  if (task_runners_.GetRasterTaskRunner() ==
+      task_runners_.GetPlatformTaskRunner()) {
+    return true;
+  }
+  if (surface_ == nullptr || raster_thread_merger_.get() == nullptr) {
+    return false;
+  }
+  fml::TaskRunner::RunNowOrPostTask(task_runners_.GetRasterTaskRunner(),
+                                    [weak_this = weak_factory_.GetWeakPtr(),
+                                     thread_merger = raster_thread_merger_]() {
+                                      if (weak_this->surface_ == nullptr) {
+                                        return;
+                                      }
+                                      thread_merger->MergeWithLease(10);
+                                    });
+  raster_thread_merger_->WaitUntilMerged();
+  return true;
```

Comment:
should this be `raster_thread_merger_->IsMerged()`?

Reply:
raster_thread_merger_->IsMerged() should always be true here because `raster_thread_merger_->WaitUntilMerged();` made sure of that. We can add a DCHeck tho.

Reply:
ah. didn't see the FML_DCHECK  :)